### PR TITLE
[NEXUS-8973] force base url property

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-8973

### CODE REVIEW

File: components/nexus-core/src/main/java/org/sonatype/nexus/internal/app/BaseUrlManagerImpl.java [hunk 1]
```diff
@@ -40,15 +40,18 @@
...
-    this.requestProvider = checkNotNull(requestProvider);
-  }
-
   private volatile String url;
 
   private volatile boolean force;
 
+  @Inject
+  public BaseUrlManagerImpl(final Provider<HttpServletRequest> requestProvider,
+                            @Named("${org.sonatype.nexus.internal.app.BaseUrlManagerImpl.force:-false}") final boolean force) {
```
Comment (peterlynch) [inline components/nexus-core/src/main/java/org/sonatype/nexus/internal/app/BaseUrlManagerImpl.java]: I never plan to tell anyone to use this property - so I guess I don't care too much that it doesn't follow the convention of nexus.feature.subfeature

Comment (jdillon) [inline components/nexus-core/src/main/java/org/sonatype/nexus/internal/app/BaseUrlManagerImpl.java]: I used this class-name style for stuff that is internal and shouldn't be fuxed with on purpose.

**Comment by mcculls** [comment]
+1

**Comment by kellyrob99** [comment]
+1

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**